src/groups/mqb/mqbs/mqbs_filebackedstorage.cpp

```diff
@@ -968,6 +1005,45 @@ void FileBackedStorage::purge(const mqbu::StorageKey& appKey)
     }
 }
 
+void FileBackedStorage::startAutoConfirming(const bmqt::MessageGUID& msgGUID)
+{
+    clearSelection();
+    d_currentlyAutoConfirming = msgGUID;
+}
+
+mqbi::StorageResult::Enum
+FileBackedStorage::autoConfirm(const mqbu::StorageKey& appKey,
+                               bsls::Types::Uint64     timestamp)
+{
+    BSLS_ASSERT_SAFE(!appKey.isNull());
+    BSLS_ASSERT_SAFE(!d_currentlyAutoConfirming.isUnset());
+
+    DataStoreRecordHandle handle;
+    int                   rc = d_store_p->writeConfirmRecord(&handle,
+                                           d_currentlyAutoConfirming,
+                                           d_queueKey,
+                                           appKey,
+                                           timestamp,
+                                           ConfirmReason::e_AUTO_CONFIRMED);
+    if (0 != rc) {
+        return mqbi::StorageResult::e_WRITE_FAILURE;  // RETURN
+    }
+    d_ephemeralConfirms.push_back(handle);
+    d_virtualStorageCatalog.autoConfirm(d_currentlyAutoConfirming, appKey);
+
+    return mqbi::StorageResult::e_SUCCESS;
+}
+
+void FileBackedStorage::clearSelection()
+{
+    for (unsigned int i = 0; i < d_ephemeralConfirms.size(); ++i) {
+        d_store_p->removeRecordRaw(d_ephemeralConfirms[i]);
```

Comment:
As discussed, it will be the responsibility of [`mqbs::FileStore::recoverMessages()`](https://github.com/bloomberg/blazingmq/blob/e888fcdef8a6d825d2d08bbd3c0d8549d3537905/src/groups/mqb/mqbs/mqbs_filestore.cpp#L1011) not [`void StorageManager::recoveredQueuesCbImpl()`](https://github.com/bloomberg/blazingmq/blob/e888fcdef8a6d825d2d08bbd3c0d8549d3537905/src/groups/mqb/mqbblp/mqbblp_storagemanager.cpp#L532) to check against stale confirm records.